CMakeLists.txt

```diff
@@ -448,3 +446,76 @@ foreach(flag IN LISTS SUNSHINE_COMPILE_OPTIONS)
 endforeach()
 
 target_compile_options(sunshine PRIVATE $<$<COMPILE_LANGUAGE:CXX>:${SUNSHINE_COMPILE_OPTIONS}>;$<$<COMPILE_LANGUAGE:CUDA>:${SUNSHINE_COMPILE_OPTIONS_CUDA};-std=c++17>)
+
+#############
+# CPACK
+####
+
+# Add all assets dependencies
+if(WIN32) # TODO: test
+	install(TARGETS sunshine RUNTIME DESTINATION ".")
+
+	install(DIRECTORY "${SUNSHINE_ASSETS_DIR}/web" DESTINATION ".")
+	install(FILES "${SUNSHINE_ASSETS_DIR}/sunshine.conf" DESTINATION ".")
+	install(FILES "${SUNSHINE_ASSETS_DIR}/apps_windows.json" DESTINATION ".")
+	install(DIRECTORY "${SUNSHINE_ASSETS_DIR}/shaders/directx" DESTINATION "shaders")
+endif()
+if(APPLE) # TODO: test
+
+	set(prefix "${CMAKE_PROJECT_NAME}.app/Contents")
+	set(INSTALL_RUNTIME_DIR "${prefix}/MacOS")
+
+	install(DIRECTORY "${SUNSHINE_ASSETS_DIR}/web" DESTINATION "${INSTALL_RUNTIME_DIR}")
+	install(FILES "${SUNSHINE_ASSETS_DIR}/sunshine.conf" DESTINATION "${INSTALL_RUNTIME_DIR}")
+
+	install(TARGETS sunshine
+			BUNDLE DESTINATION . COMPONENT Runtime
+			RUNTIME DESTINATION ${INSTALL_RUNTIME_DIR} COMPONENT Runtime)
+
+	# TODO: bundle doesn't produce a valid .app use cpack -G DragNDrop
+	#	set(CPACK_BUNDLE_NAME "Sunshine")
+	#	set(CPACK_BUNDLE_PLIST "${SUNSHINE_ASSETS_DIR}/info.plist")
+	#	set(CPACK_BUNDLE_ICON "${PROJECT_SOURCE_DIR}/sunshine.icns")
+endif()
+if(UNIX AND NOT APPLE)
+	install(DIRECTORY "${SUNSHINE_ASSETS_DIR}/web" DESTINATION ".")
+	install(FILES "${SUNSHINE_ASSETS_DIR}/sunshine.conf" DESTINATION ".")
+
+	install(FILES "${SUNSHINE_ASSETS_DIR}/apps_linux.json" DESTINATION ".")
+	install(FILES "${SUNSHINE_ASSETS_DIR}/85-sunshine-rules.rules" DESTINATION "/etc/udev/rules.d")
+	install(TARGETS sunshine RUNTIME DESTINATION "/usr/bin")
+	install(FILES "${CMAKE_CURRENT_BINARY_DIR}/sunshine.service" DESTINATION "/usr/lib/systemd/user")
+	install(DIRECTORY "${SUNSHINE_ASSETS_DIR}/shaders/opengl" DESTINATION "shaders")
+
+	# Pre and post install
+	set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA
+		"${SUNSHINE_ASSETS_DIR}/linux-deb/preinst;${SUNSHINE_ASSETS_DIR}/linux-deb/postinst;${SUNSHINE_ASSETS_DIR}/linux-deb/conffiles")
+	set(CPACK_RPM_PRE_INSTALL_SCRIPT_FILE "${SUNSHINE_ASSETS_DIR}/linux-deb/preinst")
+	set(CPACK_RPM_POST_INSTALL_SCRIPT_FILE "${SUNSHINE_ASSETS_DIR}/linux-deb/postinst")
```

Comment:
What do you think of using `post_install` and `pre_install` for the file names? I think it might make it more clear what they are, especially for non native English speaking contributors.

Also, any issue with adding `.sh` as the file extensions?

Reply:
Those names are tied to the `.deb` format but I can probably rename then in the root and bundling them correctly during the build.

Reply:
Okay, no problem then. Leaving as is, is okay too.